core/iwasm/compilation/aot_emit_memory.c

```diff
@@ -852,9 +885,27 @@ aot_compile_op_memory_grow(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx)
         }
     }
 
+#if WASM_ENABLE_MEMORY64 != 0
+    if (IS_MEMORY64) {
+        if (!(u32_max = I64_CONST(UINT32_MAX))) {
+            aot_set_last_error("llvm build const failed");
+            return false;
+        }
+        BUILD_ICMP(LLVMIntUGT, delta, u32_max, ret_value, "page_size_cmp");
+        if (!(check_page_size = LLVMAppendBasicBlockInContext(
+                  comp_ctx->context, func_ctx->func, "check_page_size"))) {
+            aot_set_last_error("llvm add basic block failed.");
+            goto fail;
+        }
+        if (!(aot_emit_exception(comp_ctx, func_ctx, EXCE_INTEGER_OVERFLOW,
+                                 true, ret_value, check_page_size)))
```

Comment:
When `memory.grow` fails, it should return -1 but not throw exception, see wasm spec and the implementation in interpreter:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_interp_classic.c#L4557

Suggest here:
```C
    BUILD_ICMP(LLVMIntUGT, delta, u32_max, ret_value1, "page_size_cmp");
```

And change below to:
```C
    BUILD_ICMP(LLVMIntUGT, ret_value, I8_ZERO, ret_value, "mem_grow_ret");

    BUILD_OP(Or, ret_value, ret_value1, ret_value, "or");  => ret_value = ret_value | ret_value1;

    /* ret_value = ret_value == true ? delta : pre_page_count */
    if (!(ret_value = LLVMBuildSelect(
              comp_ctx->builder, ret_value, mem_size,
              MEMORY64_COND_VALUE(I64_NEG_ONE, I32_NEG_ONE), "mem_grow_ret"))) {
        aot_set_last_error("llvm build select failed.");
        return false;
    }

    PUSH_PAGE_COUNT(ret_value);
```

Reply:
Sorry that the logic should be `if (delta <= u32_max && aot_enlarge_memory()) then return pre_page_count, else return -1`, so seems had better:
```C
BUILD_ICMP(LLVMIntULE, delta, u32_max, ret_value1, "page_size_cmp");
...

BUILD_ICMP(LLVMIntUGT, ret_value, I8_ZERO, ret_value, "mem_grow_ret");

#if WASM_ENABLE_MEMORY64 != 0
    BUILD_OP(And, ret_value1, ret_value, ret_value, "and");  => ret_value = ret_value1 & ret_value;
#endif
```